controllers/oooRequests.ts

```diff
@@ -102,4 +102,25 @@ export const updateOooRequestController = async (req: OooRequestUpdateRequest, r
         logger.error(ERROR_WHILE_UPDATING_OOO_REQUEST, err);
         return res.boom.badImplementation(ERROR_WHILE_UPDATING_OOO_REQUEST);
     }
+}
+
+export const getOooRequestsController = async (req: OooRequestCreateRequest, res: OooRequestResponse) => {
+    const { query } = req;
+    try {
+        const oooRequests = await getOooRequests(
+            query.type,
+            query.requestedBy,
+            query.state,
+            query.id
+        );
+        if ('error' in oooRequests) {
+            return res.status(400).json({
+                message: oooRequests.error,
+            });
+        }
+        return oooRequests;
+    } catch (err) {
+        logger.error(ERROR_WHILE_CREATING_OOO_REQUEST, err);
+        return res.boom.badImplementation(ERROR_WHILE_CREATING_OOO_REQUEST);
+    }
```

Comment:
Since this is a GET operation, wouldn’t ERROR_WHILE_FETCHING_OOO_REQUEST be a more appropriate message?

Reply:
Done